src/models/users-schema.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+const mongoose = require('mongoose');
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
+const roles = require('./roles-schema');
+
+
+
+/**
+ * The schema definition for a user record
+ * @type {mongoose.Schema}
+ */
+const users = new mongoose.Schema({
+  username: { type: String, required: true, unique: true },
+  password: { type: String, required: true },
+  email: { type: String },
+  role: { type: String, default: 'user', enum: ['admin', 'editor', 'user'] },
+},
+{ toObject: {virtuals:true}, toJSON: {virtuals: true}},
+);
+
+users.virtual('virtual_role', {
+  ref: 'roles',
+  localField: 'role',
+  foreignField: 'role',
+  justOne: true,
+
+});
+
+users.pre('findOne', function() {
+  this.populate('virtual_role');
+});
+/**
+ * this function compares a plaintext password with the stored hashed password with the stored hased
+ * password for an individual user record (`this` refers to an individual )
+ * for an individua; user record (`this` referes to an indevidual record)
+ * @return {boolean}   password match
+ * @param string paintext 
+ */
```

Comment:
are you missing the `users.pre('save', function() { this.password = bcrypt.hash(this.password, 10);` ?